syncer/syncer.go

```diff
@@ -1722,7 +1727,9 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext) e
 		onlineDDLTableNames map[string]*filter.Table
 	)
 
-	sqls, onlineDDLTableNames, err = s.resolveDDLSQL(ec.tctx, ec.parser2, parseResult.stmt, usedSchema)
+	// for DDL, we don't apply operator until we try to execute it.
+	// so can handle sharding cases
```

Comment:
🤔 should explain why use `parser.New()` here rather than `apply operator`?

Reply:
and https://github.com/pingcap/dm/pull/889/commits/cb0708180704801452a768b339b1c1f41d920f3a 😂